newm-chain-db/src/main/kotlin/io/newm/chain/database/repository/ChainRepositoryImpl.kt

```diff
@@ -187,11 +188,18 @@ class ChainRepositoryImpl : ChainRepository {
     }
 
     override fun rollbackMonitoredAddressChain(address: String, blockNumber: Long): Int = transaction {
-        MonitoredAddressChainTable.deleteWhere { MonitoredAddressChainTable.address eq address and (height greaterEq blockNumber) }
+        MonitoredAddressChainTable.deleteWhere {
+            (MonitoredAddressChainTable.address eq address) and
+                (height greaterEq blockNumber)
+        }
     }
 
     override fun pruneMonitoredAddressChainHistory(address: String, currentBlockNumber: Long): Int = transaction {
-        MonitoredAddressChainTable.deleteWhere { MonitoredAddressChainTable.address eq address and (height less currentBlockNumber - 1000) }
+        MonitoredAddressChainTable.deleteWhere {
+            (MonitoredAddressChainTable.address eq address) and
+                (height less currentBlockNumber - 1000) and
+                (height greater 0)
```

Comment:
Without this, we were wiping out the record that told us a monitored address was sync'd to the tip.